app/models/user.rb

```diff
@@ -15,6 +15,8 @@ class User < ApplicationRecord
   has_many :offers
   acts_as_taggable_on :skills
 
+  before_validation :ensure_at_least_one_tag, on: :update
```

Comment:
A theoretical question: Is there a reason why I should use `before_validation` rather than `validate`? I know they run at different times, but in this scenario it doesn't matter when we validate as long it is before we save the object, right?

Reply:
hm, you're probably right. did you try doing just `validate` ?

Reply:
I did and it also worked. But if there's no reason to choose one over the other then I will leave as is. Was just wondering if there was a reason :)